operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/KubernetesDependentResource.java

```diff
@@ -0,0 +1,146 @@
+package io.javaoperatorsdk.operator.api.reconciler.dependent;
+
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.ReconcilerUtils;
+import io.javaoperatorsdk.operator.api.config.informer.InformerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.AssociatedSecondaryResourceIdentifier;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.PrimaryResourcesRetriever;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.Mappers;
+
+public abstract class KubernetesDependentResource<R extends HasMetadata, P extends HasMetadata>
+    extends AbstractDependentResource<R, P> {
+
+  private static final Logger log = LoggerFactory.getLogger(KubernetesDependentResource.class);
+
+  protected KubernetesClient client;
+  private boolean explicitDelete = false;
+  private boolean owned = true;
+  private InformerEventSource<R, P> informerEventSource;
+
+  public KubernetesDependentResource() {
+    this(null);
+  }
+
+  public KubernetesDependentResource(KubernetesClient client) {
+    this.client = client;
+  }
+
+  protected void postProcessDesired(R desired, P primary) {
```

Comment:
Hmm not sure about this naming, it's really about filling some common data for the desired object created. But yes that maybe is better will change it. thx!

It's needed for now, there is a plan to (discuss first) to preserve labels and annotations and mainly finalizers (will actually do soon) because otherwise the update would remove this data from the resource. 

Eventually this might improve with the followup compare issue. Ideally we would do server apply just fabric8 client does not support it yet.

Reply:
Yes, I'm not too happy with that part either as this feels kind of awkward since to me this should be handle by the controller rather (i.e. this shouldn't be something that implementers of dependent resources should worry about). The controller should take these automated steps based on the dependent configuration (in this case, whether the dependent is supposed to be owned or not) and deal with owner references automatically.